src/main/java/com/epam/task/purchases/Product.java

```diff
@@ -0,0 +1,25 @@
+package com.epam.task.purchases;
+
+public class Product {
+    private final String name;
+    private final Euro price;
+
+    public Product(String name, Euro price) {
+        this.name = name;
+        this.price = price;
+    }
+
+    @Override
+    public String toString() {
+        return name + ';' + price;
```

Comment:
good! 
implicit call `toString` from `Euro`. 
and no copy-paste unlike `toString` from `AbstractPurchase`.